capabilities/service.go

```diff
@@ -0,0 +1,101 @@
+// Package capabilities provides an service to find out which capabilities/functions
+// a tenant cluster on the given installation will provide.
+package capabilities
+
+import (
+	"github.com/Masterminds/semver"
+	"github.com/giantswarm/microerror"
+
+	"github.com/giantswarm/gsctl/client"
+)
+
+// Service provides methods to get more details on the installation's
+// and tenant cluster's capabilities.
+type Service struct {
+	// Provider identifies the installation's provide,r e. g. 'aws' or 'azure'.
+	Provider string
+
+	// Client is an API client the service can use to fetch info.
+	ClientWrapper *client.Wrapper
+}
+
+// New creates a new configured Service.
+func New(provider string, clientWrapper *client.Wrapper) (*Service, error) {
```

Comment:
In general we always have `Config` struct which is then passed to `New()` function. I'm not personally super strict on this, but for general alignment that could make sense.

Reply:
Sounds a bit much to me. I can see the benefit with more arguments though.